spec/unit/pushprocessor.spec.ts

```diff
@@ -184,6 +187,24 @@ describe("NotificationService", function () {
         pushProcessor = new PushProcessor(matrixClient);
     });
 
+    it("should add default rules in the correct order", () => {
```

Comment:
I was expecting to see a call to `rewriteDefaultRules` or something here, but I now realise that is happening in the `beforeEach`.

"By the time we get here, we expect the PushProcessor to have merged the new `.m.rule.is_room_mention` rule into the existing list of rules. Check that has happened, and that it is in the right place."

Is that right? If so, can you put it in a comment?